lib/src/source_visitor.dart

```diff
@@ -108,12 +108,45 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   visitAdjacentStrings(AdjacentStrings node) {
     builder.startSpan();
-    builder.startRule();
-    visitNodes(node.strings, between: splitOrNewline);
+    var rule = new Rule();
+    builder.startRule(rule);
+
+    var chunk = zeroSplit();
+    rule.imply(chunk.rule);
+
+    visit(node.strings.first);
+    var previous = node.strings.first;
+    for (var string in node.strings.skip(1)) {
+      // Always split after a string that ends in a newline. Otherwise, indent a
+      // little bit so it's clear that the next string is a continuation rather
+      // than a new line.
+      if (_endsWithNewline(previous)) {
+        builder.writeWhitespace(Whitespace.nestedNewline);
+      } else {
+        builder.indent();
+        split();
+        builder.unindent();
+      }
```

Comment:
I think this policy is a little too specific for dartfmt. I agree it looks nice in the pub examples, but it's meaningless in lots of other examples where the strings in question are not lines of human readable text. If you look at the changed regression tests, there are strings of JSON and other stuff like that where the extra indentation is just confusing.

Also, this is what's causing the above asserts to fire. Indentation is for statement-level stuff and precedes expression nesting. Once you're inside an expression, messing with the indentation is likely to not do what you want.